src/Hamiltonians/abstract.jl

```diff
@@ -379,32 +373,38 @@ end
 #     return BasisSetRep(sm, basis, h)
 # end
 
-function BasisSetRep(h::AbstractHamiltonian, addr=starting_address(h); kwargs...)
+function BasisSetRep(h::AbstractHamiltonian, addr_or_vec=starting_address(h); kwargs...)
     # In the default case we pass `AdjointUnknown()` in order to skip the
     # symmetrisation of the sparse matrix
-    return _bsr_ensure_symmetry(AdjointUnknown(), h, addr; kwargs...)
+    return _bsr_ensure_symmetry(AdjointUnknown(), h, addr_or_vec; kwargs...)
 end
 # special cases are needed for symmetry wrappers and are defined there
 
 # default, does not enforce symmetries
 function _bsr_ensure_symmetry(
-    ::LOStructure, h::AbstractHamiltonian, addr;
+    ::LOStructure, h::AbstractHamiltonian, addr_or_vec;
     sizelim=10^6, test_approx_symmetry=true, kwargs...
 )
-    dimension(Float64, h) < sizelim || throw(ArgumentError("dimension larger than sizelim"))
-    check_address_type(h, addr)
-    sm, basis = build_sparse_matrix_from_LO(h, addr; kwargs...)
+    single_addr = addr_or_vec isa AbstractArray ? addr_or_vec[1] : addr_or_vec
+    if dimension(h, single_addr) > sizelim
+        throw(ArgumentError("dimension larger than sizelim"))
+    end
+    check_address_type(h, single_addr)
```

Comment:
Yes - but it is also a lot of work to test against all conceivable user errors. I see this more as a sanity check here - to pick up the cases where a user has mistakenly passed a non-sensical argument.

In fact, we might consider dropping this check altogether and then it would be up to the user to make sure that they pass an address that makes sense with the Hamiltonian?

Reply:
If we wanted to make the test stricter, we could test the `eltype` of the vector. I agree the address type check is not super important though and I'm fine with leaving it as is or even dropping it altogether.

Reply:
OK, I've made it check compatibility of all the passed vectors.